pyugrid/util.py

```diff
@@ -49,3 +49,28 @@ def _signed_area_tri(points):
     x3, y3 = points[2]
 
     return(((x1-x3)*(y2-y3)-(x2-x3)*(y1-y3))/2)
+
+def asarraylike(obj):
+    """
+    tests if obj acts enough like an array to be used in pyugrid. 
+
+    If it does the object is returned as is. If not, then numpy's
+    array() will be called on it.
+
+    This should catch netCDF4 variables, etc.
+
+    Note: these won't check if the attributes required actually work right.
+
+    :param obj: The object to check if it's like an array
+
+    """
+
+    must_have = ['dtype', 'shape']
```

Comment:
On Wed, Jul 15, 2015 at 4:07 PM, Filipe <notifications@github.com> wrote:

> So I say one of:
>
>
>    - what I've got now, with new methods added as we see the need or
>    - just go with array and be done with it.
>
>  I prefer the former. I am OK with adding stuff as we go.
>
It's a plan. By the way, so far we need:

numpy arrays: pass through
lists: converted t numpy arrays
netcdf4 variables: pass through

That's pretty much it -- there should be a test that netcdf4 variables do
indeed pass through.


-- 

Christopher Barker, Ph.D.
Oceanographer

Emergency Response Division
NOAA/NOS/OR&R            (206) 526-6959   voice
7600 Sand Point Way NE   (206) 526-6329   fax
Seattle, WA  98115       (206) 526-6317   main reception

Chris.Barker@noaa.gov